presto/presto.py

```diff
@@ -498,12 +498,12 @@ def forward(
         # if location overfitting is happenning
         # upd_mask[:, 0] = 1
 
-        if valid_month is not None:
+        if self.valid_month_as_token:
+            assert valid_month is not None
             val_month_token = self.valid_month_encoding(valid_month)
-            if self.valid_month_as_token:
-                x, upd_mask, orig_indices = self.add_token(
-                    val_month_token.unsqueeze(1), x, upd_mask, orig_indices
-                )
+            x, upd_mask, orig_indices = self.add_token(
+                val_month_token.unsqueeze(1), x, upd_mask, orig_indices
+            )
         else:
```

Comment:
@cbutsko why is this `else` still needed? Dummy `val_month_token ` created there is not being used anyway?